packages/p2p/src/components/modal-manager/modal-manager.tsx

```diff
@@ -1,43 +1,51 @@
 import React from 'react';
-import { modals } from 'Constants/modals';
+import { Modals, TModalKeys, TModalProps, TModals } from 'Constants/modals';
 import { useModalManagerContext } from './modal-manager-context';
+import { TModal } from './modal-manager-context-provider';
+
+type LazyModal = React.LazyExoticComponent<React.ComponentType<TModalProps[TModalKeys]>>;
```

Comment:
I had to cast the `Modal` component with the generic argument for `React.ComponentType` as `TModalProps[TModalKeys]`, otherwise it would return an interesect of all possible prop types instead of union of prop types: